posts/2021-02-15-eclipse-transformer.adoc

```diff
@@ -0,0 +1,322 @@
+
+[source]
+--
+    maven clean package
+--
+
+=== Preparing a work area
+
+Download or copy the transformer binaries to a local directory:
+
+[source]
+--
+    mkdir <work>
+    cd <work>
+    cp transformer/org.eclipse.transformer.cli/target/org.eclipse.transformer.cli-0.3.0-SNAPSHOT-distribution.jar .
+    unzip org.eclipse.transformer.cli-0.3.0-SNAPSHOT-distribution.jar
+--
+
+Unzipping the distribution obtains:
+
+[source]
+--
+    org.eclipse.transformer.cli-0.3.0-SNAPSHOT.jar
+    libs/org.eclipse.transformer-0.3.0-SNAPSHOT.jar
+    libs/biz.aQute.bnd.transform-5.2.0.jar
+    libs/slf4j-api-1.7.25.jar
+    libs/commons-cli-1.4.jar
+    libs/slf4j-simple-1.7.30.jar
+--
+
+=== Running the transformer
+
+The transformer can now be run using the default rules:
+
+[source]
+--
+  java -jar org.eclipse.transformer.cli-0.3.0-SNAPSHOT.jar \
+    <target_location> \
+    <output_location>
+--
+
+The target location can be a packed application archive file (EAR, WAR, RAR, or JAR) or a directory containing the fully or partially expanded contents of one of these archives.  The target can also be a directory containing Java source, or a directory containing several application archives.  The transformer processes all of the artifacts in the target location.  Transformed artifacts are written to the output location.
+
+(The file 'org.eclipse.transformer.cli-0.3.0-SNAPSHOT.jar' is a runnable jar file that specifies 'org.eclipse.transformer.jakarta.JakartaTransformer' as the main class and uses the necessary jars from the 'libs' folder in its class path.)
+
+The transformer can be run using custom rules by specifying additional options on the command line.  The additional options specify local rules files that the transformer uses instead of the default rules files.
+
+For example, to use Open Liberty rules data, which can be more up-to-date than the default rules data, download link:https://github.com/OpenLiberty/open-liberty/tree/integration/dev/wlp-jakartaee-transform/rules[Open Liberty Rules].  Then pull in the downloaded rules data by adding options "-tr", "-td", and "-tf", to the command line invocation:
+
+[source]
+--
+  java -jar org.eclipse.transformer.cli-0.3.0-SNAPSHOT.jar \
+    <target_location> \
+    <output_location> \
+    -tr ${rules}/jakarta-renames.properties \
+    -td ${rules}/jakarta-direct.properties \
+    -tf ${rules}/jakarta-xml-master.properties
+--
+
+The "-tr" option specifies a properties file which contains package rename data.  The "-td" option specifies a properties file which contains java direct string replacement data.  The "-tf" option specifies a master properties file which links to other properties files.  Each of the linked properties files contains replacement data for text-like files matching specific patterns, as specified in the master properties file.
```

Comment:
The "-tr" option specifies a properties file which contains
--->
The "-tr" option specifies a properties file that contains

Reply:
Update.  I also updated the other sentences in this paragraph which had this problem.